R/outbreak_step.R

```diff
@@ -90,70 +87,56 @@ outbreak_step <- function(case_data = NULL, disp.iso = NULL, disp.com = NULL,
-                                         rep(x, y)
-                                         })),
-    # draws a sample to see if this person is asymptomatic
-    asym = as.logical(rbinom(n = total_new_cases, 1, prob = prop.asym)),
-    # draws a sample to see if this person is traced
-    missed = as.logical(rbinom(n = total_new_cases, 1, prob = 1 - prop.ascertain)),
+    infector_asym = rep(asym, new_cases),
+    # cases whose parents are asymptomatic are automatically missed;
+    # will draw this for infector_asym == FALSE
+    missed = TRUE,
     # sample from the incubation period for each new person
-    incubfn_sample = inc_samples,
-    isolated = FALSE,
-    new_cases = NA
-  )
-
-
-  prob_samples <- prob_samples[exposure < infector_iso_time][, # filter out new cases prevented by isolation
-                                             `:=`(# onset of new case is exposure + incubation period sample
-                                               onset = exposure + incubfn_sample)]
-
-
-  # cases whose parents are asymptomatic are automatically missed
-  prob_samples$missed[vect_isTRUE(prob_samples$infector_asym)] <- TRUE
-
-  # If you are asymptomatic, your isolation time is Inf
-  prob_samples[, isolated_time := ifelse(vect_isTRUE(asym), Inf,
-                                        # If you are not asymptomatic, but you are missed,
-                                        # you are isolated at your symptom onset
-                                        ifelse(vect_isTRUE(missed), onset + onset_to_isolation(1),
-                                               # If you are not asymptomatic and you are traced,
-                                               # you are isolated at max(onset,infector isolation time) # max(onset,infector_iso_time)
-                                               ifelse(!vect_isTRUE(rep(quarantine, total_new_cases)),
-                                                      pmin(onset + onset_to_isolation(1), pmax(onset, infector_iso_time)),
-                                                      infector_iso_time)))]
-
+    incubfn_sample = incubation_period(total_new_cases),
+    isolated = FALSE, new_cases = NA
+  )][,
+    # draws a sample to see if this person is asymptomatic
+    asym := runif(.N) < prop.asym
+  ][
+    exposure < infector_iso_time
+  ][, # filter out new cases prevented by isolation
+      `:=`(# onset of new case is exposure + incubation period sample
+        onset = exposure + incubfn_sample
+  )]
+
+  # draw a sample for missing
+  prob_samples[infector_asym == FALSE, missed := runif(.N) > prop.ascertain]
+
+  prob_samples[, isolated_time := {
+    ref_time <- onset + onset_to_isolation(.N)
+    fifelse(
+      # If asymptomatic, never isolated: time is Inf
+      asym == TRUE, Inf, fifelse(
+      # If not asymptomatic, but are missed, isolated at your symptom onset
+      missed == TRUE, ref_time,
+      if (!is.null(quarantine) && (quarantine == TRUE)) infector_iso_time else
```

Comment:
`quarantine == TRUE` is wrapped in an extra set of parentheses, either way there is no functional difference, whichever you prefer for readability.

Reply:
yep, aware its non-functional, but I prefer this way for readability